File: GO/MostFrequentDays.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import (
	"time"
)

var DAYS = []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"}

/**
 * List form of most frequent days in order.
 */
func mostFrequentDays(year int) []string {
	var result []string

	startDay := int(time.Date(year, time.January, 1, 0, 0, 0, 0, time.UTC).Weekday())
	result = append(result, DAYS[startDay])

	if (year%4 == 0 && year%100 != 0) || year%400 == 0 {
		result = append(result, DAYS[startDay+1])
	}

	if len(result) > 1 {
		if result[0] == "Sunday" && result[1] == "Monday" {
			result[0] = "Monday"
			result[1] = "Sunday"
		}
	}

	return result
}

File: JAVA/MostFrequentDays.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.time.LocalDate;
import java.time.Month;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MostFrequentDays {

    public static String[] DAYS = {"Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday"};
    
    public static void main(String[] args) {
        System.out.println( mostFrequentDays(2427) );
    }

    /**
     * List form of most frequent days in order.
     */
    public static String[] mostFrequentDays(int year) {
        List<String> result = new ArrayList<String>();

        int startDay = LocalDate.of(year, Month.JANUARY, 1).getDayOfWeek().ordinal()+1;
        result.add( DAYS[startDay] );

        if ( (year % 4 == 0 && year % 100 != 0) || year % 400 == 0 ) {
            result.add( DAYS[startDay+1] );
        }
    
        if ( result.size() > 1 ) {
            if ( "Sunday".equals( result.get(0) ) && "Monday".equals( result.get(1) ) ) {
                Collections.reverse( result );
            }
        }

        return result.toArray(new String[0]);
    }
}
